From d710725c63e415cbd961113262bb8d3638f57d55 Mon Sep 17 00:00:00 2001
From: tpaviot <tpaviot@gmail.com>
Date: Mon, 11 Nov 2013 11:26:47 +0100
Subject: [PATCH] [osx-maverick-warning-fix][deprecated-declaration] Use
 isfinite rather than finite

Fix for following warning on OSX 10.9:
/oce/src/OSD/OSD.cxx:140:30: warning: 'finite' is deprecated: first deprecated in OS X 10.9 [-Wdeprecated-declarations]
  if ( theDivisor == 0. || ! finite(theDividend) ) return Standard_False;
                             ^
/usr/include/math.h:718:12: note: 'finite' declared here
extern int finite(double) __OSX_AVAILABLE_BUT_DEPRECATED(__MAC_10_0, __MAC_10_9, __IPHONE_NA, __IPHONE_NA);
           ^
---
 src/OSD/OSD.cxx | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

Index: oce/src/OSD/OSD.cxx
===================================================================
--- oce.orig/src/OSD/OSD.cxx
+++ oce/src/OSD/OSD.cxx
@@ -19,7 +19,9 @@
 #if HAVE_IEEEFP_H
 # include <ieeefp.h>
 #endif
-#if !defined(HAVE_FINITE) && defined(isfinite)
+#ifdef WNT
+# define finite _finite
+#elif defined(isfinite)
 # define finite isfinite
 #endif
 
@@ -77,7 +79,6 @@ Standard_Boolean OSD::CStringToReal(cons
 # define _DEXPLEN	             11
 # define _IEEE		              1
 # define DMAXEXP	             ((1 << _DEXPLEN - 1) - 1 + _IEEE)
-# define finite                      _finite
 # define SLEEP(NSEC)                 Sleep(1000*(NSEC))
 #else
 # define SLEEP(NSEC)                 sleep(NSEC)
